src/Compilers/CSharp/Test/Emit/CodeGen/SwitchTests.cs

```diff
@@ -9050,49 +9050,74 @@ public static void M(object o)
             var compVerifier = CompileAndVerify(source,
                 options: TestOptions.ReleaseDll.WithOutputKind(OutputKind.ConsoleApplication),
                 expectedOutput: "");
-            compVerifier.VerifyIL("Program.M",
-@"{
-  // Code size      106 (0x6a)
-  .maxstack  1
-  .locals init (int V_0, //i
-                object V_1)
+            compVerifier.VerifyIL("Program.M", @"
+{
+  // Code size      120 (0x78)
+  .maxstack  2
+  .locals init (int V_0,
+                int V_1, //i
+                object V_2)
   IL_0000:  ldarg.0
-  IL_0001:  stloc.1
-  IL_0002:  ldloc.1
+  IL_0001:  stloc.2
+  IL_0002:  ldloc.2
   IL_0003:  isinst     ""int""
-  IL_0008:  brfalse.s  IL_0021
-  IL_000a:  ldloc.1
+  IL_0008:  brfalse.s  IL_001c
+  IL_000a:  ldloc.2
   IL_000b:  unbox.any  ""int""
-  IL_0010:  stloc.0
+  IL_0010:  stloc.1
   IL_0011:  ldsfld     ""bool Program.b""
-  IL_0016:  brtrue.s   IL_0069
-  IL_0018:  ldsfld     ""bool Program.b""
-  IL_001d:  brtrue.s   IL_0069
-  IL_001f:  br.s       IL_002a
-  IL_0021:  ldsfld     ""bool Program.b""
-  IL_0026:  brtrue.s   IL_0069
-  IL_0028:  br.s       IL_003a
-  IL_002a:  ldsfld     ""bool Program.b""
-  IL_002f:  brtrue.s   IL_0069
-  IL_0031:  ldsfld     ""bool Program.b""
-  IL_0036:  brtrue.s   IL_0069
-  IL_0038:  br.s       IL_0043
-  IL_003a:  ldsfld     ""bool Program.b""
-  IL_003f:  brtrue.s   IL_0069
-  IL_0041:  br.s       IL_0053
-  IL_0043:  ldsfld     ""bool Program.b""
-  IL_0048:  brtrue.s   IL_0069
-  IL_004a:  ldsfld     ""bool Program.b""
-  IL_004f:  brtrue.s   IL_0069
-  IL_0051:  br.s       IL_005c
-  IL_0053:  ldsfld     ""bool Program.b""
-  IL_0058:  brtrue.s   IL_0069
-  IL_005a:  br.s       IL_0063
-  IL_005c:  ldsfld     ""bool Program.b""
-  IL_0061:  brtrue.s   IL_0069
-  IL_0063:  ldsfld     ""bool Program.b""
-  IL_0068:  pop
+  IL_0016:  brtrue.s   IL_0077
+  IL_0018:  ldc.i4.1
+  IL_0019:  stloc.0
+  IL_001a:  br.s       IL_001e
+  IL_001c:  ldc.i4.7
+  IL_001d:  stloc.0
+  IL_001e:  ldsfld     ""bool Program.b""
+  IL_0023:  brtrue.s   IL_0077
```

Comment:
These statements are repeated below, with what appears to be a dispatch to a single return location after each. Isn't this a case of a `when` expression where the evaluation should be shared with one dispatch following?